### ISSUE TITLE
BatchNormalization: option to disable trainable beta/gamma

### ISSUE THREAD

**Issue by redst4r** [fchollet/keras#4914]
Hi,

sometimes its useful to disable the centering and scaling parameters (beta/gamma) of the BatchNormalization, see e.g. [this paper, page 6](http://arxiv.org/abs/1507.02672)  (e.g. if the bn is followed by a Relu; then beta/gamma can be absorbed into the next layers weights).

In `keras.layers.BatchNormalization` there's currently no way to disable training of beta/gamma.

Tensorflow already has some convenient way to disable it in ` tf.nn.batch_normalization(x, mean, var, beta, gamma, epsilon)` by setting `beta=None` and/or `gamma=None`. This is already used in [tf.slim](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L363) via flags `center` and `scale`

We could add this fairly easily to the tensorflow backend, putting two booleans `center`, `scale` to the BatchNormalization constructor, setting e.g. `self.beta=None` in `.build()` if `self.center=False`; plus a few changes in `.call()` due to the broadcasting, which will fail on `None`

However I dont know how it would work in the theano-backend. batchnorm looks kind of more complicated there.

---
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by gvtulder** [comment]
The beta/offset/bias and gamma/scale are not currently optional for the Theano functions. Those functions are closely tied to the cuDNN implementations, and according to the manual they always require scale and bias. So I guess that you have to provide beta and gamma there. You can of course pass variables filled with zeros and ones.

It looks like `tf.nn.fused_batch_norm` also doesn't allow beta and gamma to be None, by the way. (I guess for the same cuDNN reason.)
